include/hamlib/amplifier.h

```diff
@@ -0,0 +1,475 @@
+ * enquiries about capablilities.
+ */
+
+/**
+ * Amplifier Caps
+ * \struct amp_caps
+ * \brief Amplifier data structure.
+ *
+ * The main idea of this struct is that it will be defined by the backend
+ * amplifier driver, and will remain readonly for the application.  Fields that
+ * need to be modifiable by the application are copied into the struct
+ * amp_state, which is a kind of private of the AMP instance.
+ *
+ * This way, you can have several rigs running within the same application,
+ * sharing the struct amp_caps of the backend, while keeping their own
+ * customized data.
+ *
+ * n.b.: Don't move fields around, as the backends depend on it when
+ *       initializing their caps.
+ */
+struct amp_caps
+{
+  amp_model_t amp_model;                      /*!< Amplifier model. */
+  const char *model_name;                     /*!< Model name. */
+  const char *mfg_name;                       /*!< Manufacturer. */
+  const char *version;                        /*!< Driver version. */
+  const char *copyright;                      /*!< Copyright info. */
+  enum rig_status_e status;                   /*!< Driver status. */
+
+  int amp_type;                               /*!< Amplifier type. */
+  enum rig_port_e port_type;                  /*!< Type of communication port. */
+
+  int serial_rate_min;                        /*!< Minimal serial speed. */
+  int serial_rate_max;                        /*!< Maximal serial speed. */
+  int serial_data_bits;                       /*!< Number of data bits. */
+  int serial_stop_bits;                       /*!< Number of stop bits. */
+  enum serial_parity_e serial_parity;         /*!< Parity. */
+  enum serial_handshake_e serial_handshake;   /*!< Handshake. */
+
+  int write_delay;                            /*!< Write delay. */
+  int post_write_delay;                       /*!< Post-write delay. */
+  int timeout;                                /*!< Timeout. */
+  int retry;                                  /*!< Number of retry if command fails. */
+
+  const struct confparams *cfgparams;         /*!< Configuration parametres. */
+  const rig_ptr_t priv;                       /*!< Private data. */
+
+  setting_t has_get_level;
+  setting_t has_set_level;
+
+  gran_t level_gran[RIG_SETTING_MAX]; /*!< level granularity */
+  gran_t parm_gran[RIG_SETTING_MAX]; /*!< level granularity */
+
+  /*
+   * Amp Admin API
+   *
+   */
+
+  int (*amp_init)(AMP *amp);
+  int (*amp_cleanup)(AMP *amp);
+  int (*amp_open)(AMP *amp);
+  int (*amp_close)(AMP *amp);
+
+  int (*set_freq)(AMP *amp, freq_t val);
+  int (*get_freq)(AMP *amp, freq_t *val);
+
+  int (*set_conf)(AMP *amp, token_t token, const char *val);
+  int (*get_conf)(AMP *amp, token_t token, char *val);
+
+  /*
+   *  General API commands, from most primitive to least.. :()
+   *  List Set/Get functions pairs
+   */
+
+  int (*reset)(AMP *amp, amp_reset_t reset);
+  int (*get_level)(AMP *amp, setting_t level, value_t *val);
```

Comment:
While it may be true that Elecraft amps do not support setting any additional levels, I'd like to see the API support `set_level` and especially `set_ext_level` for amps too. It's quite easy to add them now for a new API!

Reply:
Also, `get_func`/`set_func` would be useful just to be complete.